linkerd/app/core/src/classify.rs

```diff
@@ -343,4 +357,21 @@ mod tests {
             .eos(Some(&trailers));
         assert_eq!(class, Class::Grpc(Err(tonic::Code::DeadlineExceeded)));
     }
+
+    #[test]
+    fn http_handles_grpc_requests() {
+        let req = Request::builder()
+            .header(http::header::CONTENT_TYPE, "application/grpc+proto")
+            .body(())
+            .unwrap();
+        let rsp = Response::builder().status(StatusCode::OK).body(()).unwrap();
+        let mut trailers = HeaderMap::new();
+        trailers.insert("grpc-status", 4.into());
```

Comment:
should we also be testing the case where the `grpc-status` header is in the response start?